### PR TITLE
Add glint types

### PR DESCRIPTION
**Pull request by vlascik**
Types were taken from https://github.com/Gavant/glint-template-types/tree/main/types/ember-render-modifiers . 

These depend on proper export of types for class modifiers from `@ember/modifier`, which seems be missing as of yet.

### CODE REVIEW

**Review by esbanarango — APPROVED** [review]

**Review by RobbieTheWagner — APPROVED** [review]

**Comment by RobbieTheWagner** [comment]
@vlascik is there a way we can work around the `@ember/modifier` thing?

**Comment by vlascik** [comment]
> @vlascik is there a way we can work around the `@ember/modifier` thing?

As far as I can tell `@ember/modifier` now has the proper types & exports.

**Comment by vlascik** [comment]
I investigated a bit further, the issue was that the runtime of this addon imports from a package `@ember/modifier`, which doesn't really exist (might be some ember virtual package), so TS couldn't find any types for it. Real package is named `ember-modifier`, so I used that instead.

I also added the explicit dependency on `ember-modifier` and moved `ember-auto-import` to dependencies as well. The addon should now follow all the v1 addon recommendations from ember-cli-typescript and glint.

**Comment by esbanarango** [comment]
This is great!🙏

**Comment by RobbieTheWagner** [comment]
@vlascik this is awesome, thank you! It seems like CI isn't running though? I want to make sure all the tests pass before merging.

**Comment by vlascik** [comment]
Well, seems that CI is not running on other PRs on this repo either. 

This PR only adds a bunch of `.d.ts` files, and doesn't change anything related to CI (apart from updating a few packages).

FWIW `ember test` passes on my machine.

**Comment by kategengler** [comment]
@RobbieTheWagner Not exactly, but I have seen this on some repos where the scheduled workflow was suspended due to repo inactivity. It seems to affect PRs too. I've re-enabled that now. I think updating the PR may trigger CI

File: README.md
Comment (RobbieTheWagner) [inline README.md:256]:
```suggestion
 
```

Trying to trigger CI

**Comment by RobbieTheWagner** [comment]
@vlascik I got CI all green now, so could you please rebase your PR?

**Comment by vlascik** [comment]
> @vlascik I got CI all green now, so could you please rebase your PR?

I tried, but messing with a new Git client seems that I ended up with merge instead. However there was only one conflict in yarn.lock, so maybe it will pass CI anyway. I can't start it though, only maintainers can.

If CI breaks I'll try to unravel it some other way.

**Pull request closed without merging**
